File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/PlaceViewWithCancel.java
package net.cyclestreets.views;

import net.cyclestreets.view.R;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageButton;

public class PlaceViewWithCancel extends PlaceViewBase
{
  private final ImageButton cancel_;

  public PlaceViewWithCancel(final Context context) {
    this(context, null);
  }

  public PlaceViewWithCancel(final Context context, final AttributeSet attrs) {
    super(context, R.layout.placetextviewcancel, attrs);

    cancel_ = (ImageButton)findViewById(R.id.cancelBtn);
  }

  public void enableCancel(final boolean enabled) { cancel_.setEnabled(enabled); }
  public void setCancelOnClick(final OnClickListener listener) { cancel_.setOnClickListener(listener); }
}

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/place/PlaceViewWithCancel.kt
package net.cyclestreets.views.place

import net.cyclestreets.view.R
import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.ImageButton

class PlaceViewWithCancel @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) :
        PlaceViewBase(context, R.layout.placetextviewcancel, attrs) {

    private val cancel: ImageButton = findViewById(R.id.cancelBtn)

    fun enableCancel(enabled: Boolean) {
        cancel.isEnabled = enabled
    }

    fun setCancelOnClick(listener: View.OnClickListener) {
        cancel.setOnClickListener(listener)
    }
}
